index.src.html

```diff
@@ -2798,95 +2798,84 @@
+    Layout and Allocation Size=] algorithm with |parsedRect|,
+    |init|.{{VideoFrameBufferInit/format}}, and |optLayout|.
+9. If |combinedLayout| is an excpetion, throw |combinedLayout|.
+10. If `data.byteLength` is less than |combinedLayout|’s
+    [=combined buffer layout/allocationSize=], throw a {{TypeError}}.
+11. Let |resource| be a new [=media resource=] containing a copy |data|. Use
+        {{VideoFrameBufferInit/visibleRect}} and {{VideoFrameBufferInit/layout}}
+        to determine where in |data| the pixels for each plane reside.
 
     NOTE: The User Agent may choose to allocate resource with a larger coded
-      size and plane strides to improve memory alignment. Increases will be
-      reflected by {{VideoFrame/codedWidth}} and {{VideoFrame/codedHeight}}.
-
-4. Let |resourceReference| be a reference to |resource|.
-5. Let |frame| be a new {{VideoFrame}} object initialized as follows:
-    1. Assign |resourceReference| to
-            {{VideoFrame/[[resource reference]]}}.
-    2. Assign {{VideoFramePlaneInit/format}} to {{VideoFrame/[[format]]}}.
-    3. For each |planeInit| in |planes|:
-        1. Let |rows| be the number of sample rows for the Plane coresponding
-            to |planeInit| as defined by {{VideoFrame/[[format]]}}.
-        2. Let |dataOffset| be `0`.
-        3. If {{PlaneInit/offset}} [=map/exists=], assign {{PlaneInit/offset}}
-            to |dataOffset|.
-        4. Let |row| be `0`.
-        5. While |row| is less than |rows|:
-            1. Copy bytes from {{PlaneInit/data}} to |resource|, starting with
-                the byte positioned at |dataOffset| and stopping after
-                {{PlaneInit/stride}} bytes have been copied.
-
-                NOTE: The User Agent may use {{VideoFramePlaneInit/visibleRect}}
-                    to copy only the visible rectangle. It may also reposition
-                    the visible rectangle within |resource|. The final position
-                    will be reflected by {{VideoFrame/visibleRect}}.
-
-            2. Increment |dataOffset| by {{PlaneInit/stride}}.
-            3. Increment |row| by `1`.
-
-    4. Let |resourceCodedWidth| be the coded width of |resource|.
-    5. Let |resourceCodedHeight| be the coded height of |resource|.
-    6. Let |resourceVisibleLeft| be the left offset for the visible rectangle of
-        |resource|.
-    7. Let |resourceVisibleTop| be the top offset for the visible rectangle of
-        |resource|.
-
-        ISSUE: The spec should provide definitions (and possibly diagrams) for
-            coded size, visible rectangle, and display size. See
-            [#166](https://github.com/w3c/webcodecs/issues/166).
-
-    8. Assign |resourceCodedWidth|, |resourceCodedHeight|,
+        size and plane strides to improve memory alignment. Increases will be
+        reflected by {{VideoFrame/codedWidth}} and {{VideoFrame/codedHeight}}.
+        Additionally, the User Agent may use
```

Comment:
User-Agent. Also, please don't use may in non-normative parts. This probably needs to be normative. Is there any reason this is not a normative requirement? Sounds like should to me maybe, that's often desirable but maybe nto always.

Reply:
Fixed the note (made it normative). I was just confusing non-normative with normative-but-optional ("may"). 

Our existing mentions of "user agent" are without the hyphen. Looking at the html spec, I see this is far more common. Has the guidance changed to use a hyphen?

Reply:
Not sure, and now I see that quite a few specs aren't in fact consistent. Let's not care too much.